sql/core/src/test/resources/sql-tests/results/window.sql.out

```diff
@@ -1342,3 +1342,93 @@ org.apache.spark.sql.AnalysisException
     "windowName" : "w"
   }
 }
+
+
+-- !query
+create or replace temp view t1 (p, o) as values (1, 1), (1, 1), (1, 2), (2, 1), (2, 1), (2, 2)
+-- !query schema
+struct<>
+-- !query output
+
+
+
+-- !query
+select * from (select *, dense_rank() over (partition by p order by o) as rnk from t1) where rnk = 1
+-- !query schema
+struct<p:int,o:int,rnk:int>
+-- !query output
+1	1	1
+1	1	1
+2	1	1
+2	1	1
```

Comment:
It seems the output is incorrect.

Reply:
This is the correct result. I confirmed with the optimization disabled, and with postgres: https://www.db-fiddle.com/f/e74NGemZYtGkHvDwFoUaQ7/0

There are two rows with (1, 1) and two rows with (2, 1) which have rank 1 in their respective partitions.

Reply:
You can disable InferWindowGroupLimit, and find out the result.
If  postgres output as you said, could you investigate the behavior of other databases.

Reply:
I did check the results with InferWindowGroupLimit disabled, and they match (4 rows).

What makes you think the results are incorrect? There are two rows in each partition that are tied with rank 1 - and the results match that.

Reply:
@beliefer I just test this against `Spark 3.3.2`:

```
spark-sql> create or replace temp view t1 (p, o) as values (1, 1), (1, 1), (1, 2), (2, 1), (2, 1), (2, 2);
Time taken: 0.955 seconds
spark-sql> select * from (select *, dense_rank() over (partition by p order by o) as rnk from t1) where rnk = 1;
1	1	1
1	1	1
2	1	1
2	1	1
Time taken: 1.101 seconds, Fetched 4 row(s)
```

Reply:
Last day, I make InferWindowGroupLimit disabled and the output is 3 rows. Maybe I make a mistake. I will check it again.

Reply:
@beliefer `Spark 3.4.1` returns the same results as `Spark 3.3.2`.

But the master branch returns:
```
spark-sql (default)> create or replace temp view t1 (p, o) as values (1, 1), (1, 1), (1, 2), (2, 1), (2, 1), (2, 2);
Time taken: 1.148 seconds
spark-sql (default)> select * from (select *, dense_rank() over (partition by p order by o) as rnk from t1) where rnk = 1;
1       1       1
1       1       1
2       1       1
Time taken: 1.346 seconds, Fetched 3 row(s)
```

Reply:
@zhengruifeng @jchen5 After my double check, the result is OK.

Reply:
👍 Thanks for double checking